### PR TITLE
Hellfire Stew

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
None

#### Purpose of change
Adds Hellfire stew and the recipe to make it. Hellfire Stew was made by soldiers during the Civil War. Also adjusts hardtack fun to the negative since hardtack really requires at minimum some kind of liquid to make it edible.

#### Describe the solution
Adds recipe for Hellfire Stew using an old civil war recipe.   I made it autolearned because if you have hardtack you'll figure out things to do with it pretty quickly.

#### Describe alternatives you've considered
none

#### Testing
tbd

### CODE REVIEW

File: data/json/recipes/recipe_food.json [hunk 1]
```diff
@@ -2942,6 +2942,23 @@
...
+    "subcategory": "CSC_FOOD_MEAT",
+    "skill_used": "cooking",
+    "difficulty": 1,
+    "time": "20 m",
+    "autolearn": true,
+    "batch_time_factors": [ 80, 4 ],
+    "qualities": [ { "id": "COOK", "level": 3 } ],
+    "tools": [ [ [ "surface_heat", 5, "LIST" ] ] ],
+    "using": [ [ "edible_fat", 1 ] ],
+    "components": [ [ [ "hardtack", 1 ] ], [ [ "water", 1 ] ] ]
```
Comment (AtomicFox556) [inline data/json/recipes/recipe_food.json:2960]:
```suggestion
    "components": [ [ [ "hardtack", 1 ] ], [ [ "water", 1 ], [ "water_clean", 1 ] ] ]
```

File: data/json/items/comestibles/soup.json [hunk 2]
```diff
@@ -235,6 +235,28 @@
...
+    "name": { "str": "hellfire stew" },
+    "weight": "250 g",
+    "color": "brown",
+    "spoils_in": "5 days",
+    "container": "can_medium",
+    "comestible_type": "DRINK",
+    "symbol": "o",
+    "quench": 2,
+    "calories": 504,
+    "description": "An old recipe from the American Civil War.  It's edible.",
```
Comment (Fris0uman) [inline data/json/items/comestibles/soup.json:251]: That is very little description For something with hellfire in the name.

File: data/json/items/comestibles/bread.json [hunk 3]
```diff
@@ -408,7 +408,7 @@
     "flags": [ "EDIBLE_FROZEN" ],
     "vitamins": [ [ "calcium", 2 ], [ "iron", 15 ] ],
     "stack_size": 20,
-    "fun": -1
+    "fun": -4
```
Comment (Fris0uman) [inline data/json/items/comestibles/bread.json:411]: -4 is a pretty hefty penalty for something that is tasteless.

Comment (NetSysFire) [reply data/json/items/comestibles/bread.json:411]: I never ate hardtack but I can imagine it sticks to your mouth and basically dries it out. Apparently its even that unpopular people of times past have written a song about it: https://en.wikipedia.org/wiki/Hard_Tack_Come_Again_No_More

**Review by NetSysFire — COMMENTED** [review]
I know this may be out of scope, but https://www.washingtonpost.com/wp-dyn/articles/A46037-2004Dec8.html also mentions a "pudding" utilizing hard tack, would that be reasonable to add, too?

Also I wonder if hard tack should give you the disgusting rations debuff, too.

File: data/json/items/comestibles/bread.json (hunk 3 above)
Comment (Maleclypse) [reply data/json/items/comestibles/bread.json:411]: Hardtack is described as being so hard as to be unedible without some liquid and that soldiers on the march would occassionally be so desperately hungry that they would let it soak up the spit in their mouth to be soft enough to chew it so I wonder if -4 is enough.

**Comment by Maleclypse** [comment]
> I know this may be out of scope, but https://www.washingtonpost.com/wp-dyn/articles/A46037-2004Dec8.html also mentions a "pudding" utilizing hard tack, would that be reasonable to add, too?
> 
> Also I wonder if hard tack should give you the disgusting rations debuff, too.

I've moved this to draft to see about some of the options you suggest.

File: data/json/items/comestibles/soup.json (hunk 2 above)
Comment (Maleclypse) [reply data/json/items/comestibles/soup.json:251]: That it is.  I'll fix that.

File: data/json/items/comestibles/bread.json (hunk 3 above)
Comment (NetSysFire) [reply data/json/items/comestibles/bread.json:411]: Apparenlty hardtack might be easy enough to make at home. There is only one way to find out: Make some hardtack for science!

File: data/json/items/comestibles/fruit_dishes.json [hunk 4]
```diff
@@ -469,6 +469,29 @@
...
+    "calories": 100,
+    "description": "An old recipe from the American Civil War.  First you crush your hardtack into flour with whatever object you have at hand then combine it with the water and regular flour to make a dough.  You're going to want to knead in the dried fruits.  Boil up your dough into pbread pudding chunks.  Finally you add the sweetener.  Soldiers would call this a dessert.",
+    "price": 50,
+    "price_postapoc": 100,
+    "material": [ "flesh", "wheat" ],
+    "volume": "500 ml",
+    "charges": 4,
+    "phase": "liquid",
+    "flags": [ "EATEN_HOT", "USE_EAT_VERB" ],
+    "vitamins": [ [ "vitC", 2 ], [ "iron", 15 ], [ "bad_food", 2 ] ],
```
Comment (Maleclypse) [inline data/json/items/comestibles/fruit_dishes.json:493]:
```suggestion
    "vitamins": [ [ "vitC", 2 ], [ "iron", 15 ], [ "bad_food", 2 ] ]
```

**Comment by NetSysFire** [comment]
I found another recipe with hardtack in it: https://en.wikipedia.org/wiki/Fish_and_brewis

File: data/json/items/comestibles/bread.json (hunk 3 above)
Comment (remd) [reply data/json/items/comestibles/bread.json:411]: I think `-4` is a good number for consuming hardtack on its own but an ideal solution would add a recipe that allows the player to soak the hardtack in some water for 10-15 minutes to remove the Joy penalty (or return the value back to `-1`).

File: data/json/items/comestibles/fruit_dishes.json [hunk 5]
```diff
@@ -469,6 +469,29 @@
...
+    "name": { "str": "hardtack pudding" },
+    "weight": "250 g",
+    "color": "brown",
+    "spoils_in": "5 days",
+    "container": "can_medium",
+    "comestible_type": "DRINK",
+    "symbol": "o",
+    "quench": 2,
+    "calories": 100,
+    "description": "An old recipe from the American Civil War.  First you crush your hardtack into flour with whatever object you have at hand then combine it with the water and regular flour to make a dough.  You're going to want to knead in the dried fruits.  Boil up your dough into pbread pudding chunks.  Finally you add the sweetener.  Soldiers would call this a dessert.",
```
Comment (Maleclypse) [inline data/json/items/comestibles/fruit_dishes.json:485]:
```suggestion
    "description": "An old recipe from the American Civil War.  First you crush your hardtack into flour with whatever object you have at hand then combine it with the water and regular flour to make a dough.  You're going to want to knead in the dried fruits.  Boil up your dough into bread pudding chunks.  Finally you add the sweetener.  Soldiers would call this a dessert.",
```

**Comment by Maleclypse** [comment]
> I found another recipe with hardtack in it: https://en.wikipedia.org/wiki/Fish_and_brewis

Hmm I could probably make that as a transform since that's an overnight recipe?

**Comment by schoolcf** [comment]
how come the description for this is "here's how to cook it" while the description for everything else is "this is what it is"

**Comment by Maleclypse** [comment]
> how come the description for this is "here's how to cook it" while the description for everything else is "this is what it is"

I decided to try something different since these are essentially ad hoc recipes to make an unpalatable foodstuff, palatable.  If you read the instructions they do in fact describe what it is, but I personally felt that saying "Hardtack made into a bread pudding", was incredibly short and gave you the reader no clue about what hardtack made into a bread pudding would be like.

**Comment by kevingranade** [comment]
Your json seems to be crashing the tests.

**Pull request merged**